### ISSUE TITLE
zero config throwing error for missing dependency

### ISSUE THREAD

**Issue by johnpapa** [vuejs/vue-cli#2558]
### Version
3.0.3

### Reproduction link
[https://gist.github.com/johnpapa/9c7745ac9d9367a00fbd9649a74c9a4d](https://gist.github.com/johnpapa/9c7745ac9d9367a00fbd9649a74c9a4d)

### Node and OS info
8.12.0, 6.4.1

### Steps to reproduce
```
npm install -g @vue/cli @vue/cli-service-global
echo '<template><h1>Hello!</h1></template>' > App.vue
vue serve
```

### What is expected?
serve the zero config example

### What is actually happening?
```
~/_play/zeroconfig                                                             
▶vue serve       
 INFO  Starting development server...
 94% after seal                                                               /        
 ERROR  Failed to compile with 1 errors                                 21:20:03

This dependency was not found:

* ~entry in /Users/papa/.nvm/versions/node/v8.12.0/lib/node_modules/@vue/cli-service-global/template/main.js

To install it, you can run: npm install --save ~entry
```

**Comment by johnpapa** [comment]
I also tried Node.js 8.11.3, and had the same error

**Comment by LinusBorg** [comment]
Hm, seems to be related to your local npm setup.

Have you 

* checked if the packge is installed or missing in the location given in the error message? The location 
* checked if this package was already installed globally through yarn?

**Comment by johnpapa** [comment]
Thanks for the suggestion. I'm willing to consider a lot of causes. i did check and it is indeed in the right folder. i then to remove them and install again ...

```
npm uninstall -g @vue/cli-service-global
npm install -g @vue/cli-service-global
npm list -g --depth=0
```

and it does list it being there. The command does start to run, too, as we can see in the previous notes.

I navigated to the file in question at `* ~entry in /Users/papa/.nvm/versions/node/v8.12.0/lib/node_modules/@vue/cli-service-global/template/main.js`

and it seems to be where the `main.js` file is importing the `App.vue` file using the syntax ...

```
▶cd /Users/papa/.nvm/versions/node/v8.12.0/lib/node_modules/@vue/cli-service-global/template/

@vue/cli-service-global/template  93990ab ✔                              147d  
▶cat main.js 
import Vue from 'vue'
import App from '~entry' // <--- This line here

Vue.config.productionTip = false

new Vue({ render: h => h(App) }).$mount('#app')
``` 

Which I believe is a Webpack thing.

I wonder if there is a subpackage in webpack that changed or a new version somewhere. 

I also then tried using yarn instead of npm, and received the same error.

**Comment by sodatea** [comment]
Maybe related to case-sensitivity issues. A similar issue here: https://github.com/vuejs/vue-cli/issues/2305
But according to that issue this error should only be raised for `app.vue` rather than `App.vue`. So I'm not sure if you are experiencing the same bug.

Anyway, here are some possible fixes:
1. `vue serve App.vue` instead of `vue serve`
2. Remove `case-sensitive-paths-webpack-plugin` by adding these lines to `vue.config.js`:
```js
module.exports = {
  chainWebpack: config => config.plugins.delete('case-sensitive-paths')
}
```

**Comment by johnpapa** [comment]
Here is something odd ... it works in 2 other folders I tried from scratch. Check out this gif.

![2018-09-20_08-54-37](https://user-images.githubusercontent.com/1202528/45819660-eeb77080-bcb2-11e8-999a-13ec95bfc23f.gif)

**Comment by johnpapa** [comment]
Thanks @sodatea for the ideas, I tried them but they did not work.

However, I was just able to get it to work by doing this in the folder that did not work previously ...

```
rm -rf App.vue
echo '<template><h1>Hello!</h1></template>' > App.vue
vue serve
```

The only thing I can think of is that the file had a bad invisible character in it. I'm guessing here.

This is still very odd as I ran the same command that I show in the first comment (copy/paste) each time `echo '<template><h1>Hello!</h1></template>' > App.vue`

So right now I guess we can close this ... but wow, how bizarre. I wish I could explain it with certainty.

Thank you for your help. I do truly appreciate it.
